tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1657,9 +1693,19 @@ tensorflow::Status ConvertConv2DHelper(OpConverterParams* params, int group) {
     nvinfer1::DimsHW effective_kernel_size = kernel_size;
     effective_kernel_size.h() += (kernel_size.h() - 1) * (dilation.h() - 1);
     effective_kernel_size.w() += (kernel_size.w() - 1) * (dilation.w() - 1);
-    padding = CreateSamePadding(
-        stride, effective_kernel_size,
-        {static_cast<int>(tensor_dim.d[1]), static_cast<int>(tensor_dim.d[2])});
+    std::vector<int64_t> input_dims;
+    if (is_deconv) {
+      // For deconv, calculate padding based on "input_sizes" input, which
+      // actually corresponds to output size. ("input_sizes" makes sense in the
+      // context of Conv2DBackpropInput).
+      auto output_size_weights = static_cast<int*>(
+          const_cast<void*>(deconv_output_size.weights().GetValues()));
+      input_dims = {output_size_weights[h_index], output_size_weights[w_index]};
```

Comment:
let's add a comment that we are using h_index and w_index in this branch unlike {1, 2} in the else branch, because we only transpose the input and not output_size_weights.